helm/external-dns-app/templates/deployment.yaml

```diff
@@ -11,12 +11,18 @@ spec:
   selector:
     matchLabels:
       {{- include "labels.selector" . | nindent 6 }}
+      {{- if and (eq .Values.provider "gcp") (.Values.gcpServiceAccountID) (.Values.gcpWorkloadIdentityPool) (.Values.gcpIdentityProvider) }}
+      giantswarm.io/gcp-workload-identity: enabled
```

Comment:
This will probably break the rollout on existing clusters - why is this required anyway?

Reply:
Don't get me wrong: For me `giantswarm.io/gcp-workload-identity` rather sounds like a property than a criteria.

Reply:
There is an admission controller that is watching `Pods` to inject the credentials for workload identity. The admission controller only receive requests for `Pods` with this label, because annotations can't be used to select resources. So we need the label rather than an annotation.

Reply:
I think @Gacko is right. Don't change the `matchLabels`. The change below should be enough

Reply:
You can still add this label to the normal labels section but please do not add it to the selector labels. Labels like `app.kubernetes.io/name`, `app.kubernetes.io/instance` and `app.kubernetes.io/component` normally don't change because an app's name, instance or component is normally not bound to any kind of function nor describing how the app works and therefore can be used as selector labels quite well.

`giantswarm.io/gcp-workload-identity` to me sounds like a label which might change in the future or at least does not differentiate several deployments of this app. So as long as there won't be several deployments of this app with several unique identifying values other than `enabled` and required for telling one pod apart from another around, I definitely wouldn't make it a selector label.

Reply:
Ok, I removed it. Thanks a lot for the feedback.